cmd/client-fs.go

```diff
@@ -338,12 +340,61 @@ func (f *fsClient) put(reader io.Reader, size int64, metadata map[string][]strin
 			err := f.toClientError(e, objectPath)
 			return totalWritten, err.Trace(objectPartPath, objectPath)
 		}
+
+		if len(metadata["mc-attrs"]) != 0 {
+			attr := parseAttribute(metadata["mc-attrs"][0])
+
+			mode, err := strconv.ParseUint(attr["mode"], 10, 32)
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+			// Change the mode of file
+			if err := os.Chmod(objectPath, os.FileMode(mode)); err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+			uid, err := strconv.Atoi(attr["uid"])
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+			gid, err := strconv.Atoi(attr["uid"])
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+			// Change the owner
+			if err := os.Chown(objectPath, uid, gid); err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+			atime, err := strconv.ParseInt(attr["atime"], 10, 64)
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+			ctime, err := strconv.ParseInt(attr["ctime"], 10, 64)
+			if err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+			// Change the access, modify and change time
+			if err := os.Chtimes(objectPath, time.Unix(atime, 0), time.Unix(ctime, 0)); err != nil {
+				return totalWritten, probe.NewError(err)
+			}
+
+		}
 	}
 	return totalWritten, nil
 }
 
 // Put - create a new file with metadata.
 func (f *fsClient) Put(ctx context.Context, reader io.Reader, size int64, metadata map[string]string, progress io.Reader, sse encrypt.ServerSide) (int64, *probe.Error) {
+	if metadata["mc-attrs"] != "" {
+		meta := make(map[string][]string)
+		val := make([]string, 1)
```

Comment:
you don't need to do this it is ideal to have a more idiomatic style slice doesn't need allocation

```
meta["mc-attrs"] = append(meta["mc-attrs"], metadata["mc-attrs"])
```